src/main/kotlin/com/hibob/academy/employee_feedback_feature/validation/RolePermissionValidator.kt

```diff
@@ -0,0 +1,40 @@
+package com.hibob.academy.employee_feedback_feature.validation
+
+import com.hibob.academy.employee_feedback_feature.dao.EmployeeRole
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.core.Response
+
+class RolePermissionValidator {
+    companion object {
+        fun extractClaimAsLong(requestContext: ContainerRequestContext, claim: String): Long? {
+            return (requestContext.getProperty(claim) as? Number)?.toLong() ?: run {
+                requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).entity("UNAUTHORIZED: Missing or invalid $claim").build())
+                null
+            }
+        }
+
+        fun extractRole(requestContext: ContainerRequestContext): EmployeeRole? {
+            return (requestContext.getProperty("role") as? String)?.uppercase()?.let {
+                try {
+                    EmployeeRole.valueOf(it)
+                } catch (e: IllegalArgumentException) {
+                    requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).entity("UNAUTHORIZED: Invalid role").build())
+                    null
+                }
+            } ?: run {
+                requestContext.abortWith(
+                    Response.status(Response.Status.UNAUTHORIZED).entity("UNAUTHORIZED: Missing role").build()
+                )
+                null
+            }
+        }
+
+        fun hrOrAdminValidation(role: EmployeeRole): Boolean {
+            return role == EmployeeRole.ADMIN || role == EmployeeRole.HR
```

Comment:
We don't want this mechanism to be aware of why kinds of api endpoints we have, and by the names and validations that you have done here, it is very aware of it.

Also, think what happens if we need a new kind of validation, for example - isManager. It means that I have to come to this file and create a new method?